Error while checking build/O2/fullCI_slc9 for 856bf0de9f11d7a2592b8ebbd15217d47b12603f at 2026-04-10 17:18:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:564:30: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:566:38: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:568:11: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:571:30: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:573:39: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:575:11: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:584:57: error: prefer transparent functors 'greater<>' [modernize-use-transparent-functors]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:640:18: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:642:18: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:644:18: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:646:19: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:648:19: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/slc9_x86-64-slc9_x86-64/0/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx:650:20: error: statement should be inside braces [readability-braces-around-statements]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/15268/856bf0de9f11d7a2592b8ebbd15217d47b12603f/build_O2_fullCI_slc9/pretty.html).

--- a/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx
+++ b/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx
@@ -674,6 +674,7 @@ bool ITSBeamBackgroundStudy::searchBCfromMap(std::map<long, std::set<int>>& BCpe
       return true;
     }
   }
+  return false;
 }
 
 void ITSBeamBackgroundStudy::getClusterPatterns(gsl::span<const o2::itsmft::CompClusterExt>& ITSclus, gsl::span<const unsigned char>& ITSpatt, const o2::itsmft::TopologyDictionary& mdict)

--- a/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx
+++ b/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx
@@ -729,7 +729,7 @@ DataProcessorSpec getITSBeamBackgroundStudy(mask_t srcTracksMask, mask_t srcClus
                                                               dataRequest->inputs,
                                                               true);
   return DataProcessorSpec{
-    "its-zdc-anomaly-study",
+    "its-beambkg-study",
     dataRequest->inputs,
     outputs,
     AlgorithmSpec{adaptFromTask<ITSBeamBackgroundStudy>(dataRequest, ggRequest, useMC)},

--- a/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx
+++ b/Detectors/ITSMFT/ITS/postprocessing/studies/src/ITSBeamBackgroundStudy.cxx
@@ -561,19 +561,21 @@ void ITSBeamBackgroundStudy::process(o2::globaltracking::RecoContainer& recoData
 
       Tchip = ic;
 
-      if (inTFROFcounter < 1)
+      if (inTFROFcounter < 1) {
         Tmissingafter = -1;
-      else if (ChipSeenInLastROF[ic])
+      } else if (ChipSeenInLastROF[ic]) {
         Tmissingafter = 0;
-      else
+      } else {
         Tmissingafter = 1;
+      }
 
-      if (inTFROFcounter < 2)
+      if (inTFROFcounter < 2) {
         Tmissingafter2 = -1;
-      else if (ChipSeenInLast2ROF[ic])
+      } else if (ChipSeenInLast2ROF[ic]) {
         Tmissingafter2 = 0;
-      else
+      } else {
         Tmissingafter2 = 1;
+      }
 
       Tphi = ChipToPhi(ic);
 
@@ -581,7 +583,7 @@ void ITSBeamBackgroundStudy::process(o2::globaltracking::RecoContainer& recoData
       Tmincol = MAPcoo_mincol[ic];
       Tmaxcol = MAPcoo_maxcol[ic];
 
-      std::sort(MAPsize[ic].begin(), MAPsize[ic].end(), std::greater<int>());
+      std::sort(MAPsize[ic].begin(), MAPsize[ic].end(), std::greater<>());
 
       Tnhit = Tnclus_s20 = Tnclus_s100 = Tnclus_s150 = 0;
       Tnhit1 = Tnhit10 = 0.;
@@ -637,18 +639,24 @@ void ITSBeamBackgroundStudy::process(o2::globaltracking::RecoContainer& recoData
 // TODO: To be improved using geometry tools
 int ITSBeamBackgroundStudy::ChipToLayer(int chip)
 {
-  if (chip < 108)
+  if (chip < 108) {
     return 0;
-  if (chip < 252)
+  }
+  if (chip < 252) {
     return 1;
-  if (chip < 432)
+  }
+  if (chip < 432) {
     return 2;
-  if (chip < 3120)
+  }
+  if (chip < 3120) {
     return 3;
-  if (chip < 6480)
+  }
+  if (chip < 6480) {
     return 4;
-  if (chip < 14712)
+  }
+  if (chip < 14712) {
     return 5;
+  }
   return 6;
 }
 

